app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx

```diff
@@ -90,46 +93,53 @@ export default function ConnectionsTabs({
             {tabsData.data.map((connectorType, connectorTypeIndex) => (
               <ConnectionTab
                 role="button"
+                data-testid={`connectionstabs-eachtab-${connectionColumnIndex}-${connectorTypeIndex}`}
                 onClick={() => {
-                  if (index > 1) {
+                  if (connectionColumnIndex > 1) {
                     if (connectorType.canBrowse) {
-                      handleChange(connectorType, index);
+                      handleChange(connectorType, connectionColumnIndex);
                     }
                   } else {
-                    handleChange(connectorType, index);
+                    handleChange(connectorType, connectionColumnIndex);
                   }
                 }}
                 label={
-                  index > 1 ? (
+                  connectionColumnIndex > 1 ? (
                     connectorType.canBrowse ? (
                       <TabLabelCanBrowse
-                        label={connectorType.name}
-                        count={index === 0 ? connectorType.count : undefined}
-                        index={index}
+                        label={connectorType.name as string}
+                        count={
+                          connectionColumnIndex === 0 ? (connectorType.count as number) : undefined
```

Comment:
How could this ever happen `connectionColumnIndex === 0` ? the ternary above is `connectionColumnIndex > 1 ?` - also I really don't like having nested ternary operators - its much more confusing.

Reply:
Done.